core/chains/evm/logpoller/orm.go

```diff
@@ -129,3 +130,33 @@ func (o *ORM) SelectLogsByBlockRangeFilter(start, end int64, address common.Addr
 	}
 	return logs, nil
 }
+
+// LatestLogEventSigsAddrs finds the latest log by (address, event) combination that matches a list of addresses and list of events
+func (o *ORM) LatestLogEventSigsAddrs(fromBlock int64, addresses []common.Address, eventSigs []common.Hash, qopts ...pg.QOpt) ([]Log, error) {
+	var logs []Log
+
+	sigs := [][]byte{}
+	for _, sig := range eventSigs {
+		sigs = append(sigs, sig.Bytes())
+	}
+	addrs := [][]byte{}
+	for _, addr := range addresses {
+		addrs = append(addrs, addr.Bytes())
+	}
+
+	q := o.q.WithOpts(qopts...)
+	err := q.Select(&logs, `
+		SELECT max(block_number) as block_number, evm_chain_id, address, event_sig, topics, data
+		FROM logs WHERE 
+			evm_chain_id = $1 AND
+			event_sig = ANY($2) AND
+			address = ANY($3) AND
+			block_number > $4
+		GROUP BY evm_chain_id, address, event_sig, topics, data
+		ORDER BY block_number ASC
```

Comment:
I'd like to see an EXPLAIN ANALYZE of this query run on a large, real-world dataset.

Reply:
Discussed offline